astroquery/vizier/core.py

```diff
@@ -370,22 +373,35 @@ def query_region_async(self, coordinates, radius=None, inner_radius=None,
 
         # Process coordinates
         if isinstance(coordinates, (commons.CoordClasses,) + six.string_types):
-            c = commons.parse_coordinates(coordinates).transform_to('fk5')
+            c = commons.parse_coordinates(coordinates).transform_to(frame)
 
             if not c.isscalar:
                 center["-c"] = []
                 for pos in c:
-                    ra_deg = pos.ra.to_string(unit="deg", decimal=True,
-                                              precision=8)
-                    dec_deg = pos.dec.to_string(unit="deg", decimal=True,
+                    if frame == 'galactic':
+                        glon_deg = pos.l.to_string(unit="deg", decimal=True,
+                                                precision=8)
+                        glat_deg = pos.b.to_string(unit="deg", decimal=True,
                                                 precision=8, alwayssign=True)
-                    center["-c"] += ["{}{}".format(ra_deg, dec_deg)]
+                        center["-c"] += ["B{}{}".format(glon_deg, glat_deg)]
```

Comment:
I believe this `B` should be `G`
```suggestion
                        center["-c"] += [f"G{glon_deg}{glat_deg}"]
```